### PR TITLE
[PBIOS-434] Label Pill Kit

### PR DESCRIPTION
**Pull request by RachelRadford21**
**What does this PR do?** 
[PBIOS-434] Label Pill Kit

**Screenshots:**
Simulator Screenshot - iPhone 15 Pro Max - 2024-06-18 at 13 16 49

**How to test?** Steps to confirm the desired behavior:
4. See addition/change

### Checklist
- [x] **LABELS** - Add a label: `breaking`, `bug`, `improvement`, `documentation`, or `enhancement`. See [Labels](https://github.com/powerhome/playbook-apple/labels) for descriptions.
- [x] **RELEASES** - Add the appropriate label: `Ready for Testing` / `Ready for Release`
- [x] **TESTING** - Have you tested your story?

### CODE REVIEW

File: Sources/Playbook/Components/Label Pill/PBLabelPill.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+//  This software is distributed under the ISC License
+//
+//  PBLabelPill.swift
+//
+
+import SwiftUI
+
+public struct PBLabelPill: View {
+  let variant: PBPill.Variant
+  let labelValue: String?
```
Comment (isismsilva) [inline Sources/Playbook/Components/Label Pill/PBLabelPill.swift:14]: Let's remove the Value from the props name.

File: Sources/Playbook/Components/Label Pill/PBLabelPill.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  let pillValue: String?
+  let labelFontSize: PBFont
+  let labelColor: Color
+  let offset: Double?
+  let horizontalPadding: CGFloat?
+  public init(
+    variant: PBPill.Variant = .neutral,
+    labelValue: String? = "Service Needed",
+    pillValue: String? = "76",
+    labelFontSize: PBFont = .caption,
```
Comment (isismsilva) [inline Sources/Playbook/Components/Label Pill/PBLabelPill.swift:24]: I think the font is always caption. This prop is never being called. Guess we can remove that.

Comment (RachelRadford21) [reply Sources/Playbook/Components/Label Pill/PBLabelPill.swift:24]: Ok, I was thinking we could keep this so users have the choice?

File: Sources/Playbook/Components/Label Pill/PBLabelPill.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+public extension PBLabelPill {
+  var pillLabelView: some View {
+    HStack(spacing: horizontalPadding) {
+      labelView
+      pillView
+    }
+  }
+  var labelView: some View {
+    PBLabelValue(label ?? "")
```
Comment (isismsilva) [inline Sources/Playbook/Components/Label Pill/PBLabelPill.swift:45]: using a empty string will add extra size to the view. Use the safe unwrap to make this an optional view and show when the value is valid. 
Same case for the pill.

**Review by isismsilva — CHANGES_REQUESTED** [review]

File: Sources/Playbook/Components/Label Pill/PBLabelPill.swift (hunk 3 above)
Comment (RachelRadford21) [reply Sources/Playbook/Components/Label Pill/PBLabelPill.swift:45]: Why do I always forget about this!?! Fixing now

File: Sources/Playbook/Components/Label Pill/PBLabelPill.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  var pillLabelView: some View {
+    HStack(spacing: horizontalPadding) {
+      labelView
+      pillView
+    }
+  }
+  @ViewBuilder
+  var labelView: some View {
+    if let label = label {
+      if let offset = offset {
```
Comment (isismsilva) [inline Sources/Playbook/Components/Label Pill/PBLabelPill.swift:47]: is this offset always necessary? If we have a label and a nil offset, the label will be hidden.

Comment (RachelRadford21) [reply Sources/Playbook/Components/Label Pill/PBLabelPill.swift:47]: If we dont have it, it won't be lined up correctly but I can fix it so we have a label if the offset is nil(maybe)

Comment (isismsilva) [reply Sources/Playbook/Components/Label Pill/PBLabelPill.swift:47]: why don't remove the offset and use the space on the HStack?

**Review by isismsilva — APPROVED** [review]

**Review by isismsilva — APPROVED** [review]

**Pull request merged**